consensus/hotstuff/timeoutcollector/timeout_processor_test.go

```diff
@@ -0,0 +1,502 @@
+				helper.WithTCNewestQC(helper.MakeQC(helper.WithQCView(s.view-1))))),
+	)
+
+	s.Run("invalid-qc-sentinel", func() {
+		*s.validator = *mocks.NewValidator(s.T())
+		s.validator.On("ValidateQC", timeout.NewestQC).Return(model.InvalidQCError{}).Once()
+
+		err := s.processor.Process(timeout)
+		require.True(s.T(), model.IsInvalidTimeoutError(err))
+	})
+	s.Run("invalid-qc-exception", func() {
+		exception := errors.New("validate-qc-failed")
+		*s.validator = *mocks.NewValidator(s.T())
+		s.validator.On("ValidateQC", timeout.NewestQC).Return(exception).Once()
+
+		err := s.processor.Process(timeout)
+		require.ErrorIs(s.T(), err, exception)
+		require.False(s.T(), model.IsInvalidTimeoutError(err))
+	})
+	s.Run("invalid-qc-err-view-for-unknown-epoch", func() {
+		*s.validator = *mocks.NewValidator(s.T())
+		s.validator.On("ValidateQC", timeout.NewestQC).Return(model.ErrViewForUnknownEpoch).Once()
+
+		err := s.processor.Process(timeout)
+		require.False(s.T(), model.IsInvalidTimeoutError(err))
+		require.NotErrorIs(s.T(), err, model.ErrViewForUnknownEpoch)
+	})
+}
+
+// TestProcess_IncludedTCInvalid tests that TimeoutProcessor correctly handles validation errors if
+// timeout is well-formed but included TC is invalid
+func (s *TimeoutProcessorTestSuite) TestProcess_IncludedTCInvalid() {
+	timeout := helper.TimeoutObjectFixture(
+		helper.WithTimeoutObjectView(s.view),
+		helper.WithTimeoutNewestQC(helper.MakeQC(helper.WithQCView(s.view-1))),
+		helper.WithTimeoutLastViewTC(
+			helper.MakeTC(helper.WithTCView(s.view-1),
+				helper.WithTCNewestQC(helper.MakeQC(helper.WithQCView(s.view-1))))),
+	)
+
+	s.Run("invalid-tc-sentinel", func() {
+		*s.validator = *mocks.NewValidator(s.T())
+		s.validator.On("ValidateQC", timeout.NewestQC).Return(nil)
+		s.validator.On("ValidateTC", timeout.LastViewTC).Return(model.InvalidTCError{})
+
+		err := s.processor.Process(timeout)
+		require.True(s.T(), model.IsInvalidTimeoutError(err))
+	})
+	s.Run("invalid-tc-exception", func() {
+		exception := errors.New("validate-tc-failed")
+		*s.validator = *mocks.NewValidator(s.T())
+		s.validator.On("ValidateQC", timeout.NewestQC).Return(nil)
+		s.validator.On("ValidateTC", timeout.LastViewTC).Return(exception).Once()
+
+		err := s.processor.Process(timeout)
+		require.ErrorIs(s.T(), err, exception)
+		require.False(s.T(), model.IsInvalidTimeoutError(err))
+	})
+	s.Run("invalid-tc-err-view-for-unknown-epoch", func() {
+		*s.validator = *mocks.NewValidator(s.T())
+		s.validator.On("ValidateQC", timeout.NewestQC).Return(nil)
+		s.validator.On("ValidateTC", timeout.LastViewTC).Return(model.ErrViewForUnknownEpoch).Once()
+
+		err := s.processor.Process(timeout)
+		require.False(s.T(), model.IsInvalidTimeoutError(err))
+		require.NotErrorIs(s.T(), err, model.ErrViewForUnknownEpoch)
+	})
+}
+
+// TestProcess_ValidTimeout tests that processing a valid timeout succeeds without error
+func (s *TimeoutProcessorTestSuite) TestProcess_ValidTimeout() {
```

Comment:
It would be great to test more cases of a valid TO:
* a TO with `LastViewTC` for last view and QC for older view
* a TO with QC for last view, and nil `LastViewTC`
* a TO with both QC for last view, and `LastViewTC` for last view